iModelCore/ecobjects/src/ECInstance.cpp

```diff
@@ -2687,6 +2698,12 @@ struct  InstanceXmlReader
                 return InstanceReadStatus::ECSchemaNotFound;
                 }
 
+            if (ECObjectsStatus::PruneSchema == schemaStatus)
+                {
+                LOG.infov("Skipping finding of the class '%s' because its schema '%s' is being pruned.", m_className.c_str(), m_fullSchemaName.c_str());
+                return InstanceReadStatus::ECSchemaNotFound;

```

Comment:
I think this can be improved. The calling code in `iModelCore\ecobjects\src\ECCustomAttribute.cpp` checks if that return value is not success, and then goes to check if the schema is being pruned.
We already know that it is being pruned, and we could return that information here, so the caller doesn't have to repeat that check. Just add a new value InstanceReadStatus::ECSchemaPruned